src/main/java/gregtech/api/nuclear/fission/components/ControlRod.java

```diff
@@ -0,0 +1,74 @@
+package gregtech.api.nuclear.fission.components;
+
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ControlRod extends ReactorComponent {
+
+    private double weight;
+    private final boolean tipModeration;
+    private final List<Pair<FuelRod, FuelRod>> fuelRodPairs = new ObjectArrayList<>();
+
+    public ControlRod(double maxTemperature, boolean tipModeration, double thermalConductivity, double mass) {
+        super(0, maxTemperature, thermalConductivity, mass, true);
+        this.tipModeration = tipModeration;
+        this.weight = 0;
+    }
+
+    public static void normalizeWeights(ArrayList<ControlRod> effectiveControlRods) {
+        double sum = 0;
+        for (ControlRod control_rod : effectiveControlRods) {
+            sum += control_rod.weight;
+        }
+        for (ControlRod control_rod : effectiveControlRods) {
+            control_rod.weight /= sum;
+        }
+    }
+
+    public static double controlRodFactor(ArrayList<ControlRod> effectiveControlRods, double insertion) {
+        double crf = 0;
+        for (ControlRod control_rod : effectiveControlRods) {
+            if (control_rod.hasModeratorTip()) {
+                if (insertion <= 0.3) {
+                    crf += insertion / 3 * control_rod.weight;
+                } else {
+                    crf += (-11F / 7 * (insertion - 0.3) + 0.1) * control_rod.weight;
```

Comment:
should these control rod functions be displayed anywhere(tooltip or jei?)

Reply:
OK, I slightly elaborated on what exactly this function does (peak at 30% control rod insertion), but I think it's a little too in-the-weeds to get into these exact numbers.